f/assignGear/fn_assignGear.sqf

```diff
@@ -35,6 +35,14 @@ _insignia_style_CSAT = "Altis"; // Options: "Altis" | "Tanoa" | "Urban"
 _insignia_styles = [_insignia_style_NATO,_insignia_style_CSAT];
 [_unit,_typeofUnit,_insignia_styles] spawn f_fnc_assignInsignia;
 
+// ====================================================================================
+
+// Universal: assign EOD flags to engineer classes
+if (_typeofUnit in ["eng","engm"]) then {
```

Comment:
I wonder if this should also/instead be attached to the engineer trait rather than the assignGear class?

Reply:
I'd prefer not to.

Even if we still only operate on assignGeared units, just checking the traits at the end of the assignGear process, that would lead to plenty of units having EOD flags that probably shouldn't (light vehicle drivers, aircraft pilots...). If we want to look at _all_ units with Engineer traits, not just assignGeared ones, it becomes a whole other thing that I don't want to deal with.

If someone wants other assignGear classes to get flags for their particular mission, they can add the class to the eligibility array easily.

Reply:
Right, derp'ed and forgot how wide the engineer trait was / was actually thinking about the `explosiveSpecialist` trait but that probably has the same problem.

Reply:
Within assignGear, only engineers have explosiveSpecialist, so no difference there. Going outside assignGear to find other explosiveSpecialists isn't something I really want to do.

I think it's actually useful to have it not dependent on the trait, because then you can have classes which can see mines (by having a detector) but not necessarily the means to remove them, meaning the flags become more important. Not in default F3 but it's something someone might want to do.